### PR TITLE
fix: updating enterprise marketing link logic

### PR DESCRIPTION
**Pull request by christopappas**

### CODE REVIEW

File: src/lib/components/SiteFooter/index.jsx [hunk 1]
```diff
@@ -315,10 +324,22 @@ class SiteFooter extends React.Component {
...
+const enterpriseLinkShape = {
+  url: PropTypes.string,
+  queryParams: PropTypes.shape(enterpriseQueryParamsShape),
+};
+
 SiteFooter.propTypes = {
   siteName: PropTypes.string,
   siteLogo: PropTypes.node,
   marketingSiteBaseUrl: PropTypes.string,
+  enterpriseMarketingLink: PropTypes.shape(enterpriseLinkShape),
```
Comment (adamstankiewicz) [inline src/lib/components/SiteFooter/index.jsx]: You can just include the shape for both the query params and the link directly in the `propTypes` object, as opposed to creating separate variables for them. Related, I'm not too sure we need to specifically call out the query params shape.

I think just the following would suffice here:
```jsx
SiteFooter.propTypes = {
  enterpriseMarketingLink: PropTypes.shape({
    url: PropTypes.string.isRequired,
    queryParams: PropTypes.shape({}),
  }),
};
```

File: src/lib/components/SiteFooter/index.jsx [hunk 2]
```diff
@@ -341,6 +362,7 @@ SiteFooter.defaultProps = {
   siteName: null,
   siteLogo: null,
   marketingSiteBaseUrl: null,
+  enterpriseMarketingLink: {},
```
Comment (adamstankiewicz) [inline src/lib/components/SiteFooter/index.jsx]: The default here should be `null`. And then, we might want to wrap the rendering of the "edX for Business" link in a conditional such that it only displays if this prop is given to the component.

File: src/lib/components/SiteFooter/index.jsx (hunk 1 above)
Comment (christopappas) [reply src/lib/components/SiteFooter/index.jsx]: 👍

File: src/lib/components/SiteFooter/index.jsx (hunk 2 above)
Comment (christopappas) [reply src/lib/components/SiteFooter/index.jsx]: 👍

File: src/lib/components/SiteFooter/index.jsx [hunk 3]
```diff
@@ -82,6 +91,21 @@ class SiteFooter extends React.Component {
     return mobileLinks;
   }
 
+  renderBusinessMarketingListItem() {
+    const { siteName, enterpriseMarketingLink } = this.props;
+    return enterpriseMarketingLink && (
```
Comment (albemarle) [inline src/lib/components/SiteFooter/index.jsx]: Personal preference, but I'm not a huge fan of && for flow control in this manner.

File: src/lib/components/SiteFooter/index.jsx [hunk 4]
```diff
@@ -82,6 +91,21 @@ class SiteFooter extends React.Component {
...
+      <li>
+        <a href={this.renderEnterpriseMarketingSiteUrl(enterpriseMarketingLink)}>
+          <FormattedMessage
+            id="footer.site-footer.link.business"
+            defaultMessage="{siteName} for Business"
+            values={{ siteName }}
```
Comment (albemarle) [inline src/lib/components/SiteFooter/index.jsx]: Can you add a ``description`` field (usually above ``values``) with a note to help explain to the translator what's going on here?  (Assuming that "Foo for Business" is even something that gets translated... or is it more like a brand name that's always in English?)

Comment (albemarle) [reply src/lib/components/SiteFooter/index.jsx]: I realize this is existing code that you just moved... alas, movement draws attention.  ;-)

File: src/lib/components/SiteFooter/index.jsx (hunk 3 above)
Comment (christopappas) [reply src/lib/components/SiteFooter/index.jsx]: What would be your preference in this case?

```
if (enterpriseMarketingLink) {
  return (<something/>);
}
return null;
```

Comment (christopappas) [reply src/lib/components/SiteFooter/index.jsx]: I'm a JS newb btw, so I'm always looking for new ways of doing things!

File: src/lib/components/SiteFooter/index.jsx (hunk 4 above)
Comment (christopappas) [reply src/lib/components/SiteFooter/index.jsx]: Would something like `description="This is a link that will point to a business marketing page for a specified website"` suffice? (Not sure how wordy is too wordy)

Comment (albemarle) [reply src/lib/components/SiteFooter/index.jsx]: Yup, that would be fine.  Thanks!

Comment (adamstankiewicz) [reply src/lib/components/SiteFooter/index.jsx]: Awhile ago, I asked about i18n for "edX for Business" on the header in the marketing site (edx-mktg). The answer I remember getting was that brands should technically not be translated.

So, we might want to just ignore the i18n translation for the brand here anyways?

File: src/lib/components/SiteFooter/index.jsx (hunk 3 above)
Comment (adamstankiewicz) [reply src/lib/components/SiteFooter/index.jsx]: @albemarle Also curious to hear your take on why you're not a huge fan of `&&` in this manner. Seems like it's a pretty common technique for flow control?
